sdk/storage/Azure.Storage.Blobs/src/BlobServiceClient.cs

```diff
@@ -252,6 +264,7 @@ internal BlobServiceClient(Uri serviceUri, HttpPipelinePolicy authentication, Bl
                   options?.Version ?? BlobClientOptions.LatestVersion,
                   new ClientDiagnostics(options),
                   options?.CustomerProvidedKey,
+                  options?._clientSideEncryptionOptions?.Clone(),
```

Comment:
I would love all of our options classes to be immutable. I feel it's good to present in the same way as our client options classes, which are get/set properties.

Really, I'm handling this the same way we handle properties in our BlobClientOptions. I'm only keeping them grouped in an object instead of pulling each one out.